tfx/extensions/experimental/kfp_compatibility/kfp_container_component.py

```diff
@@ -0,0 +1,110 @@
+# Copyright 2020 Google LLC. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Functions for creating container components from kubeflow components."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import yaml
+from google.protobuf.json_format import ParseDict
+from tfx.components.base import base_component
+from tfx.dsl.component.experimental import executor_specs, placeholders, container_component
+from tfx.extensions.experimental.kfp_compatibility.proto import kfp_component_spec_pb2
+from tfx.types.experimental.simple_artifacts import File
+from typing import Any, Callable, Dict, Text
+
+def load_kfp_yaml_container_component(
+    component_yaml_path: Text
+) -> Callable[..., base_component.BaseComponent]:
+  """Creates a container-based component from a Kubeflow component spec.
+
+  See
+  https://www.kubeflow.org/docs/pipelines/reference/component-spec/
+
+  Example:
+    component = load_kfp_yaml_container_component(
+      "kfp_pipelines_root/components/datasets/Chicago_Taxi_Trips/component.yaml"
+    )
+
+  Args:
+    component_yaml_path: local file path of a Kubeflow Pipelines component
+                         YAML file.
+
+  Returns:
+    Container component that can be instantiated in a TFX pipeline.
+  """
+  with open(component_yaml_path) as component_file:
+    data = yaml.load(component_file, Loader=yaml.FullLoader)
+  _convert_target_fields_to_kv_pair(data)
+  component_spec = ParseDict(data, kfp_component_spec_pb2.ComponentSpec())
+  container = component_spec.implementation.container
+  name = component_spec.name
+  image = container.image
+  command = (list(map(_get_command_line_argument_type, container.command)) +
+             list(map(_get_command_line_argument_type, container.args)))
+  # TODO(ericlege): Support classname to class translation in inputs.type
+  inputs = {item.name: File for item in component_spec.inputs}
+  outputs = {item.name: File for item in component_spec.outputs}
+  parameters = {}
+  return container_component.create_container_component(
+      name, image, command, inputs, outputs, parameters
```

Comment:
nit: use kwargs style to make this more readable and less error prine (since there are multiple string type arguments)